### DISCUSSION TITLE
Multiple servers to expose metrics on the same http endpoint for prometheus to scrape.

### DISCUSSION THREAD

**Discussion by Yamunaindiran** [prometheus/prometheus#9691] [Q&A] [answered]
Is it possible to make multiple server expose their metrics on the same listening port. Similar to Node exporter 9100, when in install node exporter I can expose metrics say
1. Server1:9100
2. Server2:9100

When I try the same with my script, I'm prompted with the error "Address already in use".
I'm using the port 8000 as mentioned in the prometheus.io.... I have tried with 8003 as well.

Attached my script, I'm a rookie in python, prometheus & Networking.
[prometheus_script.txt](https://github.com/prometheus/prometheus/files/7497135/prometheus_script.txt)

**Comment by LeviHarrison** [comment, marked as answer]
The simple answer is no, it's not possible to have two applications listen on the same port on the same machine.

As for why you're getting `Address already in use` errors, it's probably because you already have an application listening on port `8000`. You can check what ports are being listened to with [`netstat -lt` on Linux](https://linuxhint.com/check_open_ports_linux/) or [`netstat -ab` on Windows](https://www.howtogeek.com/howto/28609/how-can-i-tell-what-is-listening-on-a-tcpip-port-in-windows/).

Hope this helps!

**Reply by Yamunaindiran** [reply]
Thank very much for response, Levi.
